Modules/_io/bufferedio.c

```diff
@@ -2286,23 +2293,25 @@ bufferedrwpair_traverse(rwpair *self, visitproc visit, void *arg)
 }
 
 static int
-bufferedrwpair_clear(rwpair *self)
+bufferedrwpair_clear(PyObject *op)
 {
+    rwpair *self = _rwpair_CAST(op);
     Py_CLEAR(self->reader);
     Py_CLEAR(self->writer);
     Py_CLEAR(self->dict);
     return 0;
 }
 
 static void
-bufferedrwpair_dealloc(rwpair *self)
+bufferedrwpair_dealloc(PyObject *op)
 {
+    rwpair *self = _rwpair_CAST(op);
```

Comment:
for this one I think the cast can be removed entirely? The three things that take it take PyObject* or void*.

Reply:
Unfortunately, `weakreflist` does not exist on PyObjects :(